eng/packaging.targets

```diff
@@ -29,6 +29,15 @@
     <UseIntellisenseDocumentationFile Condition="'$(UseIntellisenseDocumentationFile)' == '' and Exists('$(LibIntellisenseDocumentationFilePath)')">true</UseIntellisenseDocumentationFile>
   </PropertyGroup>
 
+  <PropertyGroup Condition="'$(ServicingVersion)' != ''">
+    <!-- Always update the package version in servicing. -->
+    <PackageVersion>$(MajorVersion).$(MinorVersion).$(ServicingVersion)</PackageVersion>
+    <IsWindowsDesktop Condition="$(WindowsDesktopCoreAppLibrary.Contains('$(AssemblyName);'))">true</IsWindowsDesktop>
+    <_AssemblyInTargetingPack Condition="'$(IsNETCoreAppSrc)' == 'true' or '$(IsAspNetCoreApp)' == 'true' or '$(IsWindowsDesktop)' == 'true'">true</_AssemblyInTargetingPack>
+    <!-- Assembly version do not get updated in non-netfx ref pack assemblies. -->
+    <AssemblyVersion Condition="'$(_AssemblyInTargetingPack)' != 'true' or '$(TargetFrameworkIdentifier)' == '.NETFramework'">$(MajorVersion).$(MinorVersion).0.$(ServicingVersion)</AssemblyVersion>
```

Comment:
Shouldn't AssemblyVersion be set outside of this file so that it also applies to reference assemblies and source generators assemblies (the latter doesn't import this file as they aren't packable)?

Reply:
> Shouldn't AssemblyVersion be set outside of this file so that it also applies to reference assemblies and source generators assemblies (the latter doesn't import this file as they aren't packable)?

we don`t need to update the reference assemblies as we never change them in servicing and anyway we are not shipping them in 6.0.
For generators assemblies, we will have to decide if we want to always increment with the package or only when we make a change in generator. 

> Shouldn't AssemblyVersion be set outside of this files

Instead i will suggest to add the packaging.targets file to analyzers as well. (using IsNETCoreAppAnalyzer property) 


are any of the analyzers part of the shared framework ?

Reply:
> are any of the analyzers part of the shared framework ?

Yes, the System.Text.Json one is part of the shared framework.

> Instead i will suggest to add the packaging.targets file to analyzers as well. (using IsNETCoreAppAnalyzer property)

Analyzers aren't packable themselves so it doesn't make sense for them to import packaging.targets. The code in that file is specific to projects which pack. I would keep the PackageVersion property in packaging.props and move the AssemblyVersion out and make sure that it for now only applies to source projects which are packable.

> For generators assemblies, we will have to decide if we want to always increment with the package or only when we make a change in generator.

My gut feeling says that we should keep the AssemblyVersion of source assemblies in sync with their source generators. For that we might need two more properties in `NetCoreAppLibrary.props` (btw we should change that file's name at some point): `AspNetCoreAppAnalyzer` and `NetCoreAppAnalyzer`.

Reply:
Putting more thought into this, I don't think that introducing properties for analyzers in NetCoreAppLibrary.props is necessary. It is probably easier to apply the same assembly version to both the source assembly and the analyzer assembly.

If the project is a [generator project](https://github.com/dotnet/runtime/blob/3b40c7cfc4519a95e9d03ac85a2a2de68299aa7f/src/libraries/Directory.Build.props#L9) you would need to check if the "corresponding" source assembly is part of any shared framework.

Reply:
For now all existing source generators are shipping as part of a shared framework (either netcoreapp or aspnetcoreapp) so if this becomes to complicated to implement you might just ignore them for 6.0 and never increment their assembly version.

If you want to go down that path then your PR in its current state is probably sufficient.